portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -111,25 +111,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.getWriter().println(gson.toJson(LoadAllBlogsOrLast(/*all=*/false)));   
   }
  
-  private List<BlogMessage> putRepliesWithPosts(List<BlogMessage> blogMessages) {
+  private List<BlogMessage> putRepliesWithPosts(List<BlogMessage> blogMessagesAll) {
     // Separate posts from replies from |blogMessages|.
     // Add replies to messageReplies for the respective posts.
+    List<BlogMessage> blogMessagesParents = new ArrayList<BlogMessage>();
     List<BlogMessage> blogMessagesReplies = new ArrayList<BlogMessage>();
-    for (BlogMessage message : blogMessages) {
-      if (message.getParentID() != 0) {
-        blogMessagesReplies.add(message);
-        blogMessages.remove(message);
+    for (BlogMessage message : blogMessagesAll) {
+      if (message.getParentID() == 0) {
+        blogMessagesParents.add(message);
+      } else {
+          blogMessagesReplies.add(message);
       }
     }
  
-    for (BlogMessage post : blogMessages) {
+    for (BlogMessage post : blogMessagesParents) {
```

Comment:
Is there a data structure you could here that would eliminate the need for a double for loop?

Reply:
ooh.. i could probably use a hashtable and make the keys the id's making it more efficient.